src/Microsoft.DotNet.Interactive/KernelExtensions.cs

```diff
@@ -163,8 +165,8 @@ public static T UseDotNetVariableSharing<T>(this T kernel)
                 if (kernel.ParentKernel is { } composite)
                 {
                     return composite.ChildKernels
-                                    .OfType<DotNetKernel>()
-                                    .SelectMany(k => k.GetVariableNames());
+                                    .OfType<ISupportGetValue>()
+                                    .SelectMany(k => k.GetValueDescriptors().Select(vd => vd.Name)).ToArray();
```

Comment:
Seems like which kernel it's declared in will be useful here.

Reply:
now I got this :)